MovingBlocks/Terasology
# feat: Particle pool rendering

### Contains

This PR improves the rendering of particle pools.
Instead of rendering each particle individually, all particles in a particle pool are rendered at once.
The particle quads are generated on the GPU using a geometry shader. The rendering result should be practically identical to the previous implementation.

I added support for geometry shaders in `GLSLShader`. The class demanded version 1.2 shaders to be used. Since my implementation relies on OpenGL 3.3, I also added support for arbitrary shader versions. 1.2 is still used as the default in case no version is provided in a shader.
Note that the current implementation requires the version definition ("#version xxx") to be the very first line of the shader file.

I turned `ParticlePool` into a "drawable", which means it can now be rendered like a mesh. Note that this makes the class dependent on OpenGL.

### How to test

Destroy some blocks and verify that the block particles are being rendered as expected. You can also use the WeatherManager module and take a look at the weather particles.
